### PR TITLE
Don't force 2FA for OneLogin or SimplSaml

### PR DESCRIPTION
**Pull request by joshbetz**
These popular SSO solutions already provide a way for users to login
via multi-factor authentication, we don't need to add yet another
authentication mechanism to the mix.

### CODE REVIEW

**Comment by joshbetz** [comment]
Also adds a `wpcom_vip_two_factor_prep_hide_admin_notice` filter to site owners can hide the prep notice.

```
add_filter( 'wpcom_vip_is_two_factor_forced', '__return_false' );
add_filter( 'wpcom_vip_two_factor_prep_hide_admin_notice', '__return_true' );
```

I originally wanted to use the `wpcom_vip_is_two_factor_forced` filter for this purpose, but we're already using it to decide if 2FA is enabled. If we switch that to true, 2FA is forced on now, which allows site owners to opt-in to 2FA early.

File: two-factor.php [hunk 1]
```diff
@@ -27,6 +27,24 @@ function wpcom_vip_is_two_factor_forced() {
 		return false;
 	}
 
+	// Don't force 2FA for OneLogin SSO
+	if ( function_exists( 'is_saml_enabled' ) && is_saml_enabled() ) {
+		return false;
+	}
+
+	// Don't force 2FA for SimpleSaml
+	if ( function_exists( '\HumanMade\SimpleSaml\instance()' ) ) {
```
Comment (mjangda) [inline two-factor.php]: For this, is it simply enough that that method exists? Or do we need to actually check if it's enabled?

Comment (joshbetz) [reply two-factor.php]: Good point. It looks like we need to bail if the instance function return falsey, which indicates there's not a valid SAML config.

https://github.com/wpcomvip/skyscanner/blob/940e0be2dcf2fc9891db4fbe0ae96057d827cc6a/client-mu-plugins/wp-simple-saml/inc/namespace.php#L156-L159

**Comment by joshbetz** [comment]
r138198-deploy

**Comment by mjangda** [comment]
👍

**Pull request merged**